game-core/src/test/java/games/strategy/net/MessengerIntegrationTest.java

```diff
@@ -158,12 +159,14 @@ public void testMultipleClientToClient() {
 
   @Test
   public void testMultipleMessages() throws Exception {
-    final Thread t1 = new Thread(new MultipleMessageSender(serverMessenger));
-    final Thread t2 = new Thread(new MultipleMessageSender(client1Messenger));
-    final Thread t3 = new Thread(new MultipleMessageSender(client2Messenger));
+    final CountDownLatch testReadyLatch = new CountDownLatch(1);
+    final Thread t1 = new Thread(new MultipleMessageSender(serverMessenger, testReadyLatch));
+    final Thread t2 = new Thread(new MultipleMessageSender(client1Messenger, testReadyLatch));
+    final Thread t3 = new Thread(new MultipleMessageSender(client2Messenger, testReadyLatch));
     t1.start();
     t2.start();
     t3.start();
+    testReadyLatch.countDown();
```

Comment:
Did this test really fail because of the removed call to `Thread#yield`?
If yes, that's really unintuitive.

Reply:
The test didn't fail, but, putting on my Mind Reading Cap, I believe the original intention of the author was to ensure that each worker thread broadcast messages concurrently, as much as possible, to test if the messenger implementation was thread-safe.  Without any kind of synchronization, I suspect the original author saw that the first worker thread finished sending most of its messages before the second one started, and so on.

So, the purpose of the latch is to ensure all three workers at least start at the same time.  Assuming the thread scheduler is fair, they should run at approximately the same rate.  However, as you said, there's no guarantee.

Upon further review, I realized I made a mistake here and needed to add a second latch to ensure the three worker threads do indeed start simultaneously.  I added that in c461a52a2.

Reply:
I just merged the changes, but when I think about it: One latch should be enough when every worker counts down the latch and then awaits it.

Reply:
Yes, you're correct.  For some reason, I implemented the more general case assuming the main thread had to do something before the workers can run.  That's obviously not the case here.  I removed the unnecessary latch in #4400.